### PR TITLE
USHIFT-1053: remove user data dir

### PR DESCRIPTION
**Pull request by dhellmann**
Remove the user data directory and always use the global path.

### CODE REVIEW

File: pkg/config/node.go [hunk 1]
```diff
@@ -36,13 +36,13 @@ func (c *Config) CanonicalNodeName() string {
...
-func (c *Config) establishNodeName() (string, error) {
+func (c *Config) establishNodeName(dataDir string) (string, error) {
 	name := c.CanonicalNodeName()
-	filePath := filepath.Join(GetDataDir(), ".nodename")
+	filePath := filepath.Join(dataDir, ".nodename")
 	contents, err := os.ReadFile(filePath)
 	if os.IsNotExist(err) {
 		// ensure that dataDir exists
-		os.MkdirAll(GetDataDir(), 0700)
+		os.MkdirAll(DataDir, 0700)
```
Comment (eggfoobar) [inline pkg/config/node.go:45]: This is looking great, small nit, I think this needs to be the lowercase `dataDir`.

Comment (dhellmann) [reply pkg/config/node.go:45]: good eye, yes, you're right

**Pull request merged**